.github/workflows/release.yml

```diff
@@ -0,0 +1,203 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        description: 'Release type'     
+        required: true
+        default: 'auto' 
+        type: choice
+        options:
+          - minor
+          - patch 
+      repository:
+        description: Repository (in owner/repo format)
+        required: true
+        default: kiali/helm-charts
+        type: string
+      release_branch:
+        description: Branch to release
+        required: true
+        default: master
+        type: string
+  
+jobs:
+  initialize:
+    name: Initialize
+    runs-on: ubuntu-20.04
+    outputs:            
+      release_version: ${{ steps.release_version.outputs.release_version }}      
+      next_version: ${{ steps.next_version.outputs.next_version }}
+      branch_version: ${{ steps.branch_version.outputs.branch_version }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.releasing_branch }}       
+      
+      - name: Determine release version
+        env:
+          RELEASE_TYPE: ${{ github.event.inputs.release_type }}
+        id: release_version
+        run: |
+          RAW_VERSION=$(sed -rn 's/^VERSION \?= v(.*)/\1/p' Makefile)
+          
+          # Remove any pre release identifier (ie: "-SNAPSHOT")
+          RELEASE_VERSION=$(./.github/workflows/util/semver bump release $RAW_VERSION)
+          
+          # If we are doing a patch release, the Makefile contains a version that is already released.
+          # The version present in the Makefile needs to be bumped and that's our releasing version.
+          # The next version is two patches from whatever is present in the Makefile.          
+          if [[ $RELEASE_TYPE == "patch" ]]
+          then
+              RELEASE_VERSION=$(./.github/workflows/util/semver bump $RELEASE_TYPE $RELEASE_VERSION)              
+          fi
+
+          echo "::set-output name=release_version::v$RELEASE_VERSION"      
+      
+      - name: Determine next version
+        env:
+          RELEASE_TYPE: ${{ github.event.inputs.release_type }}
+          RELEASE_VERSION: ${{ steps.release_version.outputs.release_version }}
+        id: next_version
+        run: |
+          # Drop the version prefix ("v")
+          RELEASE_VERSION=${RELEASE_VERSION:1}
+
+          if [[ $RELEASE_TYPE == "patch" ]]
+          then
+              NEXT_VERSION=v$(./.github/workflows/util/semver bump $RELEASE_TYPE $RELEASE_VERSION)
```

Comment:
Wouldn't it better to simply use this line replacing the whole if/else?

And I echo that I wanted to remove the dependency on that `semver` script :-(

Reply:
I didn't know you wanted to remove, I was thinking on removing it a few days ago, but finally didn't do it, well... until now, what do you think of the following:

https://github.com/kiali/kiali/pull/4980/files#diff-4d71906b252fbf817a4b47220031fb7a35caaeb33785596cc8e5b0b392357dd3

It is based on the groovy function "bump" that the operator is currently using.

Reply:
I replaced the semver file with this new approach on the server release pipeline, I'm thinking on doing the same thing on the rest of the pipelines, is that ok for you?.